architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/architecturemodelgenerator/core/model/EntryPoint.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.architecturemodelgenerator.core.model;
+
+import io.ballerina.architecturemodelgenerator.core.model.service.DisplayAnnotation;
+import io.ballerina.architecturemodelgenerator.core.model.service.FunctionParameter;
+import io.ballerina.architecturemodelgenerator.core.model.service.Interaction;
+
+import java.util.List;
+
+/**
+ * Represents EntryPoint related information.
+ *
+ * @since 2201.2.2
+ */
+public class EntryPoint extends ModelElement {
```

Comment:
Instead of saying just `EntryPoint`, wouldn't it make more sense to refer it as `FunctionEntryPoint` as otherwise we'll have to include services as well under this new entity?

Reply:
Lets modify the generated model as well accordingly? (`entrypoint` -> `functionentrypoint`)